js/views/modals/Settings/Page.js

```diff
@@ -30,9 +30,41 @@ export default class extends baseVw {
       'click .js-avatarRight': 'avatarRightClick',
       'click .js-headerLeft': 'headerLeftClick',
       'click .js-headerRight': 'headerRightClick',
+      'change .js-primaryColorPicker': 'handleColorChosen',
+      'change .js-secondaryColorPicker': 'handleColorChosen',
+      'change .js-textColorPicker': 'handleColorChosen',
+      'change .js-highlightColorPicker': 'handleColorChosen',
+      'change .js-highlightTextColorPicker': 'handleColorChosen',
+      'change .js-primaryColorCode': 'handleColorCodeEntered',
+      'change .js-secondaryColorCode': 'handleColorCodeEntered',
+      'change .js-textColorCode': 'handleColorCodeEntered',
+      'change .js-highlightColorCode': 'handleColorCodeEntered',
+      'change .js-textHightlightColorCode': 'handleColorCodeEntered',
     };
   }
 
+  /** Handles when a hex color code is entered by updating color picker. */
+  handleColorCodeEntered(event) {
+    const colorPickerId = $(event.target).data('color-picker-id');
+    const $colorPicker = $(`${colorPickerId}`);
+    const newHexColorCode = event.target.value;
+
+    // If the text passes a basic RegExp for a valid 6 digit hex value,
+    // update the color picker's color.
+    if (/[0-9A-F]{6}/i.test(newHexColorCode)) {
+      $colorPicker.val(newHexColorCode);
+    }
+  }
+
+  /** Handles when a color is chosen from the color picker by updating hex color code text. */
+  handleColorChosen(event) {
+    const hexInputId = $(event.target).data('hex-input-id');
+    const $hexInputEl = $(`${hexInputId}`);
```

Comment:
This doesn't work (pre-fixes) because there was no "#" in that string, so that's why I was using the string templates to add it. The fix just adds it to the data properties because that's way more efficient.